src/plugins/discover/public/embeddable/saved_search_embeddable.tsx

```diff
@@ -426,12 +422,11 @@ export class SavedSearchEmbeddable
       { columns: this.input.columns || this.savedSearch.columns },
       this.services.core.uiSettings
     ).columns;
+    searchProps.sort = this.getSort(
+      this.input.sort || this.savedSearch.sort,
```

Comment:
Thanks for fixing this case!

I noticed one weird thing when a saved search configuration on Dashboard has legacy format for a column `"sort":["extension","desc"]`. It replaces the current sorting setting when trying to add another sort criteria on Dashboard (on Discover pages it works as expected)

![Aug-08-2022 10-39-45](https://user-images.githubusercontent.com/1415710/183378846-9857c4c9-e7a3-4e91-8f7f-3f9481b0874f.gif)
![Aug-08-2022 10-41-54](https://user-images.githubusercontent.com/1415710/183378855-e925a4e3-3266-40c8-9b90-a3fe608987e3.gif)

Is it something we can improve?

Reply:
definitely!  another good catch 🎣 ! thx!

Reply:
this is interesting, because it worked for me locally ... maybe you could export the dashboard saved object and send it to me so I could reproduce? many thx!

Reply:
thx for the saved object! So the problem here is that `extension` is a field that's not sortable, `extension.keyword` is sortable. when changing the sort, this field is removed from sorting (sorted out 😃). Since the sorting state of the UI can only be changed with manually tweaking the saved search, I think we're safe here. However there might be a potential follow up taking care of the behavior in Discover, which seems to work differently in this case.